### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Ghilain**
### Create custom methods

Your models should include some additional methods.

  - Users
        - A method that returns the 3 most recent posts for a given user.
  - Posts
        - A method that updates the posts counter for a user.
        - A method which returns the 5 most recent comments for a given post.
  - Comments
        - A method that updates the comments counter for a post.
  - Likes
        - A method that updates the likes counter for a post.
        - Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ActiveRecord::Base
+  has_many :posts, class_name: 'Post'
+  has_many :comments, class_name: 'Comment'
+  has_many :likes, class_name: 'Like'
```
Comment (rloterh) [inline app/models/user.rb]: - [x] As per the given _ERD_, the _User_ **association** should have _author_id_ referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve this aspect, similar to the code example below:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class Comment < ActiveRecord::Base
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] OPTIONAL
It's redundant to include `class_name: 'Post'`  to _comment_ association with _post_. Simply do something like:
```ruby
# app/model/comment.rb
belongs_to :post

File: app/models/like.rb [hunk 3]
```diff
@@ -1,2 +1,10 @@
 class Like < ActiveRecord::Base
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
```
Comment (rloterh) [inline app/models/like.rb]: - [x] OPTIONAL
It's redundant to include `class_name: 'Post'`  to _like_ association with _post_. Simply do something like:
```ruby
# app/model/like.rb
belongs_to :post

File: app/models/post.rb [hunk 4]
```diff
@@ -1,2 +1,13 @@
 class Post < ActiveRecord::Base
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+  after_save :update_posts_counter
```
Comment (rloterh) [inline app/models/post.rb:5]: - [x] OPTIONAL
Good job for invoking `after_save` method on `update_posts_counter` method, however kindly mark `update_posts_counter` as private after the invoke, similar to :
```ruby
  # app/models/post.rb
  after_save :update_posts_counter

  private

  def update_posts_counter
    author.increment!(:posts_counter)
  end

File: app/models/like.rb [hunk 5]
```diff
@@ -1,2 +1,10 @@
...
+
+  after_save :update_likes_counter
```
Comment (rloterh) [inline app/models/like.rb:5]: - [x] OPTIONAL
Good job for invoking `after_save` method on `update_likes_counter` method, however kindly mark `update_likes_counter` as private after the invoke, similar to :
```ruby
  # app/models/like.rb
  after_save :update_likes_counter

  private

  def update_likes_counter
    author.increment!(:likes_counter)
  end

File: app/models/comment.rb [hunk 6]
```diff
@@ -1,2 +1,9 @@
...
+  after_save :update_comments_counter
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] OPTIONAL
Good job for invoking `after_save` method on `update_comments_counter` method, however kindly mark `update_comments_counter` as private after the invoke, similar to :
```ruby
  # app/models/comment.rb
  after_save :update_comments_counter

  private

  def update_comments_counter
    author.increment!(:comments_counter)
  end

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ghilain, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Nice job for implementing `after_save` method on update methods


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ghilain** [comment]
thank you @rloterh

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @Ghilain 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ghilain** [comment]
Thank you @emmanuellekamwa

**Pull request merged**